build/includes/sdk.mk

```
@@ -22,12 +22,14 @@
 
 build_sdk_base_version = $(call sha,$(build_path)/build-sdk-images/tool/base/Dockerfile)
 build_sdk_base_tag = agones-build-sdk-base:$(build_sdk_base_version)
+build_sdk_version = $(call sha_bash,$(build_path)/build-sdk-images/*/*.sh)
```

A couple of questions on this:

1. What if I change the `/build-sdk-images/go/Dockerfile` then this won't rebuild, since it's only looking at .sh files, no?
1. If i change `/build-sdk-images/go/gen.sh` - won't that mean that all images will get rebuilt?

Should this be (or something close to):
```suggestion
build_sdk_version = $(call sha_bash,$(build_path)/build-sdk-images/$(SDK_FOLDER)/*)
```

?

---

Nice questions, I don't expect these scripts to be changed too often.
Currently we have rebuild every time. So this version is better, but not absolutely. Let me check both questions in practise. But I think:
1. We would have another base-image so rebuild would be triggered.

---

Sorry I'm confused :confused: , if I _only_ change `/build-sdk-images/go/Dockerfile` - then the image won't rebuild, it would only change if I changed the .sh files. This could be really confusing for the developer.

I can't see a path that would cause that to do anything else? But I could be missing something.

---

You are right definitely, let's concat those two hashes. Will resend new version in a moment.

---

I think we might be saying two different, but important things.

I think you are referencing when the `/build-sdk-images/tool/base/Dockerfile` changes.

I'm talking about when the language specific Dockerfile changes, for example `/build-sdk-images/go/Dockerfile` 

You are right - both are actually issues!

---

Just looking at this again, does this bring us back to the original problems?